src/main/groovy/com/readytalk/gradle/plugins/ci/CiLifecyclePlugin.groovy

```diff
@@ -0,0 +1,85 @@
+package com.readytalk.gradle.plugins.ci
+
+import com.readytalk.gradle.plugins.ci.tasks.CiTask
+import com.readytalk.gradle.util.PluginUtils
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.testing.Test
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+class CiLifecyclePlugin implements Plugin<Project>, PluginUtils {
+  static final String CI_TASK = 'ci'
+  static final String PUBLISH_TASK = 'publish'
+  private CiInfoExtension infoExt
+  Project project
+
+  void apply(final Project project) {
+    this.project = project
+    project.plugins.with {
+      apply(CiInfoPlugin)
+      apply(CiPublishingPlugin)
+    }
+    this.infoExt = project.plugins.getPlugin(CiInfoPlugin).extension
+    setupLifecycleTask()
+    applyConventions()
+  }
+
+  private void setupLifecycleTask() {
+    project.with {
+      tasks.create(CI_TASK, CiTask).configure { ciTask ->
+        //defaultTasks is a plain List, so we can't hook it with all{}
+        afterEvaluate {
+          if (tasks.findByName('build') != null) {
```

Comment:
Fun fact: I originally looked for the build task using `tasks.matching { it.name == 'build' }`. In newer versions of Gradle however, using matching doesn't count as asking gradle for a reference to that task name, which means the task might never be created. Whereas using `tasks.findByName` *will* lazily instantiate.